File: df702/src/main/java/edu/monash/humanise/smartcity/Payload.java
package edu.monash.humanise.smartcity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Payload {
    @Id
    @SequenceGenerator(
            name = "df702_id_sequence",
            sequenceName = "df702_id_sequence"
    )
    @GeneratedValue(
            strategy = GenerationType.SEQUENCE,
            generator = "df702_id_sequence"
    )
    private Integer id;
    @Column(nullable = false)
    private String deviceName;
    @Column(nullable = false)
    private String data;
    @Column(nullable = false)
    private String devEUI;
    @Column(nullable = false)
    private boolean battery;
    @Column(nullable = false)
    private boolean isFull;
    @Column(nullable = false)
    private boolean fire;
    @Column(nullable = false)
    private int level;
    @Column(nullable = false)
    private int temperature;

    @Column(nullable = false)
    private OffsetDateTime timestamp;

}

File: athom-presence-sensor/src/main/kotlin/edu/monash/humanise/smartcity/athompresencesensor/payload/Payload.kt
package edu.monash.humanise.smartcity.athompresencesensor.payload

import jakarta.persistence.*
import java.time.OffsetDateTime

/**
 * Name for the global ID sequence.
 */
const val SEQUENCE_NAME = "athom_presence_sensor_id_sequence"

/**
 * Presence sensor data entity.
 *
 * Each sensor type has its own entity classes, such as [OccupancyPayload] for occupancy data.
 * This class is the superclass of all sensor data classes, and does not actually exist in the database.
 */
@MappedSuperclass
open class Payload(
    /**
     * Device name of where this datapoint is from. This corresponds to the `esphome.name` attribute in the device's
     * yml file.
     */
    @Column(nullable = false)
    val deviceName: String,
    /**
     * Timestamp of datapoint according to MQTT timestamp, or when the message arrived at the collector if the timestamp is not available.
     */
    @Column(nullable = false)
    val timestamp: OffsetDateTime,
    /**
     * Raw data of the MQTT payload, stored as a string.
     */
    @Column(nullable = false)
    val data: String
) {
    /**
     * A unique number of each sensor datapoint, shared across all sensor entities. This is just a sequence that
     * counts up and does not have any other semantic meaning.
     */
    @Id
    @SequenceGenerator(name = SEQUENCE_NAME, sequenceName = SEQUENCE_NAME)
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = SEQUENCE_NAME)
    var id: Long = 0
}